File: solver333/Utility.fs
module Utility

#if !FABLE_COMPILER
open System.IO
#endif

let level = 0 // 0 = beginner, 1 = intermediate, 2 = advanced, 3 = god
let mutable cornerOrientationLevel = 0
let mutable cornerPermutationLevel = 0
let mutable fullCmll = false
let mutable edgeOrientationLevel = 0
let mutable lbPairLevel = 0
let mutable lfPairLevel = 0
let mutable rbPairLevel = 0
let mutable rfPairLevel = 0
let mutable orientCentersWithSecondBlock = false
let mutable chooseShortestSecondBlockPairOrder = false
let mutable chooseShortestFirstBlockPairOrder = false
let mutable x2yColorNeutral = false
let mutable useEolr = false

let readPatterns matchFn method level name cornerRotationNeutral cornerColorNeutral discoverAuf =
#if FABLE_COMPILER
    PatternData.read $"{method}/{level}/{name}"
#else
    File.ReadLines $"Patterns/{method}/{level}/{name}.txt"
#endif
    |> Seq.map (fun line -> line.Split ',')
    |> Seq.map (fun x ->
        let pattern = x[0]
        let algs = if x[1].Length = 0 then [] else x[1].Split ';' |> List.ofSeq
        matchFn, name, (pattern, cornerRotationNeutral, cornerColorNeutral, discoverAuf), algs)
    |> List.ofSeq

File: site/lab/solver/Utility.js
import { createAtom } from "./fable_modules/fable-library-js.4.16.0/Util.js";
import { empty, ofSeq } from "./fable_modules/fable-library-js.4.16.0/List.js";
import { map } from "./fable_modules/fable-library-js.4.16.0/Seq.js";
import { item } from "./fable_modules/fable-library-js.4.16.0/Array.js";
import { split } from "./fable_modules/fable-library-js.4.16.0/String.js";
import { read } from "./PatternData.js";

export const level = 0;

export let cornerOrientationLevel = createAtom(0);

export let cornerPermutationLevel = createAtom(0);

export let fullCmll = createAtom(false);

export let edgeOrientationLevel = createAtom(0);

export let lbPairLevel = createAtom(0);

export let lfPairLevel = createAtom(0);

export let rbPairLevel = createAtom(0);

export let rfPairLevel = createAtom(0);

export let orientCentersWithSecondBlock = createAtom(false);

export let chooseShortestSecondBlockPairOrder = createAtom(false);

export let chooseShortestFirstBlockPairOrder = createAtom(false);

export let x2yColorNeutral = createAtom(false);

export let useEolr = createAtom(false);

export function readPatterns(matchFn, method, level_1, name, cornerRotationNeutral, cornerColorNeutral, discoverAuf) {
    return ofSeq(map((x) => [matchFn, name, [item(0, x), cornerRotationNeutral, cornerColorNeutral, discoverAuf], (item(1, x).length === 0) ? empty() : ofSeq(split(item(1, x), [";"], void 0, 0))], map((line) => split(line, [","], void 0, 0), read(`${method}/${level_1}/${name}`))));
}

